sklearn/linear_model/_logistic.py

```diff
@@ -728,19 +729,19 @@ def _logistic_regression_path(X, y, pos_class=None, Cs=10, fit_intercept=True,
     if multi_class == 'multinomial':
         # scipy.optimize.minimize and newton-cg accepts only
         # ravelled parameters.
-        if solver in ['lbfgs', 'newton-cg']:
+        if solver in ['lbfgs', 'trust-ncg', 'newton-cg']:
             w0 = w0.ravel()
         target = Y_multi
-        if solver == 'lbfgs':
+        if solver in ['lbfgs', 'trust-ncg']:
             def func(x, *args): return _multinomial_loss_grad(x, *args)[0:2]
         elif solver == 'newton-cg':
             def func(x, *args): return _multinomial_loss(x, *args)[0]
             def grad(x, *args): return _multinomial_loss_grad(x, *args)[1]
-            hess = _multinomial_grad_hess
+        hess = _multinomial_grad_hess
         warm_start_sag = {'coef': w0.T}
     else:
         target = y_bin
-        if solver == 'lbfgs':
+        if solver in ['lbfgs', 'trust-ncg']:
```

Comment:
We can have another `elif` here to set the hess for the `multi_class != "multinomial"` case.